File: week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

#get str
str = get_string("Text: ")
#length
l = 0
#word
w = 1
#sentence
s = 0

#loop for chars
for i in range(len(str)):
    #" " char
    if str[i] == " ":
        w += 1
    #. char
    if str[i] == ".":
        s += 1
    #! char
    if str[i] == "!":
        s += 1
    #?char
    if str[i] == "?":
        s += 1
    #normal char
    else:
        l += 1
#kelime/harf sayısı
kelimeharf = (l*100)/w
#cumle/kelime sayısı
cumlekelime = (s*100)/w
#formula
sonuc = (0.0588*kelimeharf)-(0.296*cumlekelime)-15.8
#round and -6.2
sonuc = round(sonuc-6.2)
if sonuc > 15:
    #grater
    print("Grade 16+")
if sonuc < 1:
    #lower
    print("Before Grade 1")
if sonuc < 16:
    if sonuc > 0:
        print("Grade", sonuc)

File: week2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string txt=get_string("Text:");//girdi
    int l=0;//uzunluk
    int s=0;//cumle
    int w=1;//kelime
    for(int i=0;i<strlen(txt);i++){
        if((char) txt[i]==46||(char) txt[i]==33||(char) txt[i]== 63){//".,!,?" ise
            s++;}// cumle sayisi
        if((char) txt[i]==32){//bosluk
            w++;}
        if((char) txt[i]<=122&&(char) txt[i]>=65){//harf
            l++;
        }
    }
    double kelime_harf_orani=((float)l/(float)w)* (float) 100;//formul
    double kelime_cumle_orani=((float)s/(float)w)* (float) 100;
    double sonuc=(0.0588*kelime_harf_orani) - (0.296 *kelime_cumle_orani)- 15.8;
    int rounded=round(sonuc);
    //printf(" %i %i %i\n",l,s,w);
    printf("%f\n",sonuc);
    //printf("%i\n",rounded);
    if(rounded<=1){printf("Before Grade 1\n");}
    if(rounded>=16){printf("Grade 16+\n");}
    if(rounded>1&&rounded<16){printf("%s%i\n","Grade ",rounded);}
}